Mathlib/SetTheory/Game/Basic.lean

```diff
@@ -396,21 +399,19 @@ theorem quot_mul_comm (x y : PGame.{u}) : (⟦x * y⟧ : Game) = ⟦y * x⟧ :=
 theorem mul_comm_equiv (x y : PGame) : x * y ≈ y * x :=
   Quotient.exact <| quot_mul_comm _ _
 
-instance isEmpty_mul_zero_leftMoves (x : PGame.{u}) : IsEmpty (x * 0).LeftMoves := by
-  cases x
-  exact instIsEmptySum
-
-instance isEmpty_mul_zero_rightMoves (x : PGame.{u}) : IsEmpty (x * 0).RightMoves := by
+instance isEmpty_leftMoves_mul (x y : PGame.{u})
+    [IsEmpty (x.LeftMoves × y.LeftMoves ⊕ x.RightMoves × y.RightMoves)] :
+    IsEmpty (x * y).LeftMoves := by
   cases x
-  apply instIsEmptySum
-
-instance isEmpty_zero_mul_leftMoves (x : PGame.{u}) : IsEmpty (0 * x).LeftMoves := by
-  cases x
-  apply instIsEmptySum
+  cases y
+  assumption
```

Comment:
Do you even need the `cases x; cases y`?

Reply:
Yes, for whatever reason statements like `(mk xl xr xL xR + mk yl yr yL yR).LeftMoves = xl ⊕ yl` are def-eq, while `(x + y).LeftMoves = x.LeftMoves ⊕ y.LeftMoves` aren't.

Reply:
That's quite weird. Eta for structures should have fixed this... unless it's something to do with well-founded recursion being irreducible by default now? I am confused